core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/lockbased_txmgr.go

```diff
@@ -155,7 +154,7 @@ func (txmgr *LockBasedTxMgr) ValidateAndPrepare(blockAndPvtdata *ledger.BlockAnd
 
 	block := blockAndPvtdata.Block
 	logger.Debugf("Validating new block with num trans = [%d]", len(block.Data.Data))
-	batch, txstatsInfo, err := txmgr.validator.ValidateAndPrepareBatch(blockAndPvtdata, doMVCCValidation)
+	batch, txstatsInfo, err := txmgr.commitBatchPreparer.ValidateAndPrepareBatch(blockAndPvtdata, doMVCCValidation)
```

Comment:
Ah, I see now. Yes, I mentioned offline that we can certainly separate the code for batch preparation and validation. But that's much bigger change and it won't be of the tone `Validate(blockAndPvtdata)`. It would be more like batch preparation code is maintained in the txmgr package and validation function looks like... `Validate(tx, batchSoFar, statedb)`. But perhaps, this is not the right place to discuss the details of a potential future approach.